src/modelinstance.cpp

```diff
@@ -659,22 +658,11 @@ void ModelInstance::loadCompiledModelPtr(const plugin_config_t& pluginConfig) {
 
 plugin_config_t ModelInstance::prepareDefaultPluginConfig(const ModelConfig& config) {
     plugin_config_t pluginConfig = config.getPluginConfig();
-    // Do not add CPU_THROUGHPUT_AUTO when performance hint is specified.
-    bool isPerformanceHintSpecified = pluginConfig.count("PERFORMANCE_HINT") > 0;
-    if (isPerformanceHintSpecified) {
+    // By default, set "PERFORMANCE_HINT" = "THROUGHPUT";
+    if (!pluginConfig.empty()) {
```

Comment:
Previous behavior:
If performance hint was specified, we did not change plugin config for the user.
If performance hint was NOT specified, we set number of streams to auto.

Current behavior (your changes):
If ANY option is provided to plugin config (even not related to streams) we DO NOT set performance hint to tput. Is that desired?

I think we should add `pluginConfig["PERFORMANCE_HINT"] = "THROUGHPUT";` always when user does set NUM_STREAMS. It would match previous behavior.